packages/nimble-components/src/select/template.ts

```diff
@@ -131,7 +136,7 @@ SelectOptions
                         class="scrollable-region">
                         <slot
                             ${slotted({
-                                filter: (n: Node) => n instanceof HTMLElement && isListboxOption(n),
+                                filter: (n: Node) => n instanceof HTMLElement && (isListboxOption(n) || slottedOptionGroupFilter(n)),
```

Comment:
Is there a reason not to call this new function `isOptionGroup` to parallel the `isListboxOption` function already being called here?